benchmark/path/basename.js

```diff
@@ -0,0 +1,26 @@
+var common = require('../common.js');
+var path = require('path');
+var v8 = require('v8');
+
+var bench = common.createBenchmark(main, {
+  type: ['win32', 'posix'],
+  n: [1e6],
+});
+
+function main(conf) {
+  var n = +conf.n;
+  var p = path[conf.type];
+
+  // Force optimization before starting the benchmark
+  p.basename('/foo/bar/baz/asdf/quux.html');
+  v8.setFlagsFromString('--allow_natives_syntax');
+  eval('%OptimizeFunctionOnNextCall(p.basename)');
+  p.basename('/foo/bar/baz/asdf/quux.html');
```

Comment:
Nice :-) Did you check if v8 is really going to optimize the function? If the function is never going to be optimized by v8, then we don't have to use this, right?